File: LeetCode/101对称二叉树.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        Stack<TreeNode> sta = new Stack<>();
        if (root == null) return true;
        sta.add(root.left);
        sta.add(root.right);
        while (!sta.isEmpty()) {
            TreeNode leftNode = sta.peek();
            sta.pop();
            TreeNode rightNode = sta.peek();
            sta.pop();
            if (leftNode == null && rightNode == null) continue;
            if (leftNode == null || rightNode == null || leftNode.val != rightNode.val) return false;
            sta.add(leftNode.left);
            sta.add(rightNode.right);
            sta.add(leftNode.right);
            sta.add(rightNode.left);
        }
        return true;
    }
}

File: LeetCode/101对称二叉树.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func isSymmetric(root *TreeNode) bool {
    var que []*TreeNode
    if root == nil {
        return true
    }
    que = append(que, root.Left)
    que = append(que, root.Right)
    for len(que) > 0 {
        leftNode := que[0]
        rightNode := que[1]
        que = que[2:]
        if leftNode == nil && rightNode == nil {
            continue
        }
        if leftNode == nil || rightNode == nil || leftNode.Val != rightNode.Val {
            return false
        }
        que = append(que, leftNode.Left, rightNode.Right, leftNode.Right, rightNode.Left)
    }
    return true
}